### ISSUE TITLE
Fix str + repr of multi-slice Devices

### ISSUE THREAD

**Issue by skye** [jax-ml/jax#26418]
I think multi-slice TPU devices should still be called `TpuDevice` instead of `MegaScalePjRtDevice`. You can already tell the device is part of a multi-slice config because it has a slice_id. It's very verbose right now:

```python
In [2]: d = jax.devices()[0]

In [3]: str(d)
Out[3]: 'MegaScalePjRtDevice(wrapped=TPU_0(process=0,(0,0,0,0)), slice_id=0)'

In [4]: repr(d)
Out[4]: 'MegaScalePjRtDevice(wrapped=TpuDevice(id=0, process_index=0, coords=(0,0,0), core_on_chip=0), slice_id=0)'
```

**Comment by achaljhawar** [comment]
Hi, I am interested in working on this issue.

**Comment by aakashv000** [comment]
@skye - may i pick this up?

**Comment by ruasnv** [comment]
Hi! I am interested in solving this issue. This seems like a perfect first issue. Can I work on this?

**Comment by skye** [comment]
I don't have bandwidth to help someone learn how to fix this issue, but if anyone feels empowered to figure it out themselves, please free free to open a PR :) It think it might involve a change to PJRT in https://github.com/openxla/xla and not jax itself. It might also be hard to test without access to TPUs. (Maybe this isn't a great first issue...)

**Comment by IamRam3** [comment]
I would have definitely tried this if I had TPU's with me. My bad!

**Comment by ruasnv** [comment]
Unfortunately I don't have TPU's as well! I'm on my way to find a better first-issue. Maybe we can edit the labels for this one since it is tricky as a first-issue
